### PR TITLE
vim: add vim_darwin package for macOS

### PR DESCRIPTION
**Pull request by gurjeet**
The default Vim package does not have some advanced features like
python3, hence we need to use the vim_configurable package and customize
it to provide a fuller build of Vim on macOS.

If one were to use the vim_configurable package as is, the user does not
get the nice OS integration features like copy-pasting from system
clipboard.

###### Motivation for this change
Get a fully-featured (python etc.) Vim on macOS, along with the ability to
copy/paste from/to macOS clipboard.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -24695,6 +24695,20 @@ in
...
+    config = {
+      vim = {
+        gui = "none";
+        darwin = true;
+      };
+    };
+  }).overrideAttrs (oldAttrs: rec {
+    pname = "vim_darwin";
+    meta = {
+      platforms = [ "x86_64-darwin" ];
```
Comment (prusnak) [inline pkgs/top-level/all-packages.nix:24708]: Shouldn't this be `platforms = platforms.darwin;`? to cover `aarch64-darwin` in the future

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -24695,6 +24695,20 @@ in
     gtk3 = if stdenv.isDarwin then gtk3-x11 else gtk3;
   });
 
+  vim_darwin = (vim_configurable.override {
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:24698]:
```suggestion
  vim-darwin = (vim_configurable.override {
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -24695,6 +24695,20 @@ in
...
+    config = {
+      vim = {
+        gui = "none";
+        darwin = true;
+      };
+    };
+  }).overrideAttrs (oldAttrs: rec {
+    pname = "vim_darwin";
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:24706]:
```suggestion
    pname = "vim-darwin";
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 108109` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vim_darwin</li>
  </ul>
</details>

**Comment by gurjeet** [comment]
@prusnak @SuperSandro2000 thanks for the feedback! I have added a commit based on your feedback. Please review.

**Comment by gurjeet** [comment]
@prusnak, @SuperSandro2000, a gentle reminder to follow-up. Please let me know if I need to do something more, or ping someone else to get this merged and closed.

**Comment by prusnak** [comment]
> Please let me know if I need to do something more, or ping someone else to get this merged and closed.

I think you should ping vim maintainers

**Comment by gurjeet** [comment]
Thanks for the pointer @prusnak!

@lovek323 @equirosa, I see you listed as the maintainers in vim/common.nix. Can you please review this small PR and see if it's ready for merge.

**Review by lovek323 — APPROVED** [review]
LGTM

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by gurjeet** [comment]
A gentle reminder for merging the PR. /cc @equirosa

**Pull request merged**